### DISCUSSION TITLE
Allow instant, single-click time tracking

### DISCUSSION THREAD

**Discussion by vic-t** [kimai/kimai#2570] [General]
**TL;DR** It would be nice if Kimai gave us the possibility to start tracking time instantly and fill in the details later.

I came to Kimai from other closed-source solutions. My concerns were the same as you'd have with any other closed source software, such as no guarantee of survival, very limited options for configuration, no self-hosting options and, considering all these disadvantages, a relatively high pricing for what you get in return. 

In terms of general attractiveness of the web app, management and configuration options, Kimai is IMO better than any of these solutions. 

I do miss, however, the flexibility and simplicity some of these commercial tools offered when creating new time entries. In Kimai, the user is required to first select a project as well as an activity before time tracking can even start which in turn requires that both the project and the activity already exist. At least the way I work, that's simply not possible so that most of the time I end up writing time down on a piece of paper before I enter it into Kimai. 

I'm not going to list all the advantages these tools offer but instead just mention the one killer feature I would love to see in Kimai: The ability to create a new time entry without having to set all fields in advance. 

For example, if I get a call, I'll just hit the start button, time tracking starts with an empty entry. During _or after_ the call I may decide to add more information to the entry. Maybe it's a new activity, so I'll go create the activity, come back to the incomplete timesheet entry and fill out the missing parts. Maybe the call is not worth tracking so I'll just discard it entirely. 

Needless to say, the entry would not be activated or assigned until it is complete, i.e. until at least both a project and an activity were chosen. 

I understand this is all easier said than done. To not introduce too much complexity, maybe an idea would be to create a hidden project/activity pair that will be used to store such incomplete entries. The entries themselves can easily be searched and filtered for but they will not appear in any calculation until they have been assigned to another project/activity pair.

**Comment by kevinpapst** [comment]
> I understand this is all easier said than done. To not introduce too much complexity, maybe an idea would be to create a hidden project/activity pair that will be used to store such incomplete entries. 

Why don't you do that right now?
If you always work like that, then you will have this entry always in the "latest activities" dropdown in the upper right.
Restarting from this menu will only copy customer/project/activity and start the entry right away.

Kimai is primarily a project-time tracking software and therefor the data model requires a project.

I am keen to remove requirement for an activity in the future, but I am not sure about the project.

**Reply by vic-t** [reply]
> Why don't you do that right now?
> If you always work like that, then you will have this entry always in the "latest activities" dropdown in the upper right.

That's not exactly true. I mean, ultimately, the goal is to rectify all incomplete entries. And as soon as you did that, the "special" entry will no longer be in the latest activities.

**Reply by kevinpapst** [reply]
Hm, that is somehow true :D 

There are issues out there which describe what you want. Please have a look if you find them and vote there with a 👍

**Reply by vic-t** [reply]
> Kimai is primarily a project-time tracking software and therefor the data model requires a project.

I understand where you're coming from. You maybe have 3-5 projects you're working on at a time and you probably don't even have timesheet entries under 30 minutes. Your work is unlikely tied to phone calls but rather to your own schedule. Kimai is perfectly suited for this. 

For me, and from what I've seen in the discussions and issues, many others, it's a slightly different scenario. While we do have such project work as well, a large part is remote support. We take the call, help our customer and only then start documenting everything we've discussed and done. The call itself may be only a couple minutes, documentation will take more than that.

Unfortunately, we may get into the situation that we have to take another call even before we finished documenting the first one. Right now, we solve this by just having a piece of paper ready where we write down our times but it would be so much cleaner if Kimai allowed us empty entries so that we at least have a clean tracking of our time. 

One other major benefit of allowing empty entries is that, in the longer term, it would even allow us to connect Kimai directly to our phone system. A call can then start the entry because no other information would be required. Such information can be added during or after the call. 

> I am keen to remove requirement for an activity in the future, but I am not sure about the project.

I think this is another topic entirely and yes, I agree that 3 levels (customer, project, activity) may be too much for some. Frankly, I don't need 3 levels, that's why all my projects are called "Services for [CustomerName]". It doesn't bother me. 

That's why I wish to reiterate that my goal is not that you get rid of the requirement. It's just about creating a special purpose type of timesheet entry that can be incomplete but would also be considered inactive by Kimai.

There is an open issue asking for instant timesheet creation, see https://github.com/kevinpapst/kimai2/issues/2215. It's quite specific in its description which is why I had decided a more open feature request would be justified. Still, combining his request and your input so far, it may be an idea to have this second "call" button that would immediately start tracking time in always the same placeholder customer/project/activity combo. Although not at all elegant, this seems quite simple and would indeed solve the issue. 

What do you think?

**Reply by vic-t** [reply]
Just referencing issue https://github.com/kevinpapst/kimai2/issues/403 which I guess is related in some way if we wanted to keep things very simple.

**Reply by kevinpapst** [reply]
I appreciate the feedback! The best is to vote for #2215 - spreading the same topic across multiple places makes it harder for me when coming back to the topic.
